src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -266,6 +268,68 @@ public async Task<Document> ApplyCodeFixesForSpecificDiagnosticIdAsync(Document
             return solution.GetDocument(document.Id) ?? throw new NotSupportedException(EditorFeaturesResources.Removal_of_document_not_supported);
         }
 
+        private bool TryGetWorkspaceFixersMap(Document document, out Lazy<ImmutableDictionary<DiagnosticId, ImmutableArray<CodeFixProvider>>> fixerMap)
+        {
+            if (_lazyWorkspaceFixersMap == null)
+            {
+                var workspaceFixersMap = _getWorkspaceFixersMap(document.Project.Solution.Workspace);
+                Interlocked.CompareExchange(ref _lazyWorkspaceFixersMap, workspaceFixersMap, null);
+            }
+
+            return _lazyWorkspaceFixersMap.TryGetValue(document.Project.Language, out fixerMap);
+        }
+
+        private bool TryGetWorkspaceFixersPriorityMap(Document document, out Lazy<ImmutableDictionary<CodeFixProvider, int>> fixersPriorityMap)
+        {
+            if (_lazyFixerPriorityMap == null)
+            {
+                var fixersPriorityByLanguageMap = _getFixerPriorityMap(document.Project.Solution.Workspace);
+                Interlocked.CompareExchange(ref _lazyFixerPriorityMap, fixersPriorityByLanguageMap, null);
+            }
+
+            return _lazyFixerPriorityMap.TryGetValue(document.Project.Language, out fixersPriorityMap);
+        }
+
+        private bool TryGetWorkspaceFixer(
+            Lazy<CodeFixProvider, CodeChangeProviderMetadata> lazyFixer,
+            Workspace workspace,
+            bool showInfoBarOnException,
+            [NotNullWhen(returnValue: true)] out CodeFixProvider? fixer)
+        {
+            try
+            {
+                fixer = lazyFixer.Value;
+                return true;
+            }
+            catch (Exception ex)
+            {
+                // Gracefully handle exceptions in creating fixer instance.
+                // Show info bar for exception, if needed.
+                if (showInfoBarOnException)
+                {
+                    var errorReportingService = workspace.Services.GetRequiredService<IErrorReportingService>();
+                    var message = lazyFixer.Metadata.Name != null
+                        ? string.Format(EditorFeaturesResources.Error_creating_instance_of_CodeFixProvider_0, lazyFixer.Metadata.Name)
+                        : EditorFeaturesResources.Error_creating_instance_of_CodeFixProvider;
+
+                    errorReportingService.ShowErrorInfoInActiveView(
+                        message,
+                        new InfoBarUI(
+                            WorkspacesResources.Show_Stack_Trace,
+                            InfoBarUI.UIKind.HyperLink,
+                            () => errorReportingService.ShowDetailedErrorInfo(ex), closeAfterAction: true));
+
+                    foreach (var errorLogger in _errorLoggers)
```

Comment:
shouldn't we log this even if `showInfoBarOnException` is set to `false`?

Reply:
Let me rename the variable - we don't want to log exception or show info bar when invoked from GetFixableDiagnosticIDs context.